tests/components/twitch/test_twitch.py

```diff
@@ -0,0 +1,174 @@
+"""The tests for an update of the Twitch component."""
+from unittest.mock import MagicMock, patch
+
+from requests import HTTPError
+from twitch.resources import Channel, Follow, Stream, Subscription, User
+
+from homeassistant.components import sensor
+from homeassistant.setup import async_setup_component
+
+ENTITY_ID = "sensor.channel123"
+CONFIG = {
+    sensor.DOMAIN: {
+        "platform": "twitch",
+        "client_id": "1234",
+        "channels": ["channel123"],
+    }
+}
+CONFIG_WITH_OAUTH = {
+    sensor.DOMAIN: {
+        "platform": "twitch",
+        "client_id": "1234",
+        "channels": ["channel123"],
+        "oauth_token": "9876",
+    }
+}
+
+USER_ID = User({"id": 123, "display_name": "channel123", "logo": "logo.png"})
+STREAM_OBJECT_ONLINE = Stream(
+    {
+        "channel": {"game": "Good Game", "status": "Title"},
+        "preview": {"medium": "stream-medium.png"},
+    }
+)
+CHANNEL_OBJECT = Channel({"followers": 42, "views": 24})
+OAUTH_USER_ID = User({"id": 987})
+SUB_ACTIVE = Subscription({"created_at": "2020-01-20T21:22:42", "is_gift": False})
+FOLLOW_ACTIVE = Follow({"created_at": "2020-01-20T21:22:42"})
+
+
+async def test_init(hass):
+    """Test initial config."""
+
+    channels = MagicMock()
+    channels.get_by_id.return_value = CHANNEL_OBJECT
+    streams = MagicMock()
+    streams.get_stream_by_user.return_value = None
+
+    twitch_mock = MagicMock()
+    twitch_mock.users.translate_usernames_to_ids.return_value = [USER_ID]
+    twitch_mock.channels = channels
+    twitch_mock.streams = streams
+
+    with patch(
+        "homeassistant.components.twitch.sensor.TwitchClient", return_value=twitch_mock
+    ):
+        assert await async_setup_component(hass, sensor.DOMAIN, CONFIG) is True
+
+    sensor_state = hass.states.get(ENTITY_ID)
+    assert sensor_state.state == "offline"
+    assert sensor_state.name == "channel123"
+    assert sensor_state.attributes["icon"] == "mdi:twitch"
+    assert sensor_state.attributes["friendly_name"] == "channel123"
+    assert sensor_state.attributes["views"] == 24
+    assert sensor_state.attributes["follower"] == 42
+
+
+async def test_offline(hass):
+    """Test offline state."""
+
+    twitch_mock = MagicMock()
+    twitch_mock.users.translate_usernames_to_ids.return_value = [USER_ID]
+    twitch_mock.channels.get_by_id.return_value = CHANNEL_OBJECT
+    twitch_mock.streams.get_stream_by_user.return_value = None
```

Comment:
I am no expert at tests but I do not think you need to mock them in every single test method.
Comparing these tests with for example [rmvtransport tests](https://github.com/home-assistant/home-assistant/blob/dev/tests/components/rmvtransport/test_sensor.py).

Reply:
Can you be a little more precise? I'm not getting the point :sweat_smile:

Reply:
On further investigation I have also found tests that use this sort of patter, so I think it is fine 😅